### PR TITLE
chore: removed DatasetArtifactType

### PR DESCRIPTION
**Pull request by metakuni**
### Reviewers
**Functional:** 
@Bento007 

**Readability:** 
@danieljhegeman 

---

## Changes
- remove: unused `DatasetArtifactType` enum, `dataset_artifact.type` column

### CODE REVIEW

File: backend/Makefile [hunk 1]
```diff
@@ -80,11 +80,12 @@ else
 endif
 
 db/test_migration:
-	$(MAKE) db/dump_schema > /tmp/before
+	$(MAKE) db/dump_schema > /tmp/before_migration
 	$(MAKE) db/migrate
+	$(MAKE) db/dump_schema > /tmp/after_migration
```
Comment (metakuni) [inline backend/Makefile:85]: Added this to make it easier to check the schema after migration.

File: backend/Makefile [hunk 2]
```diff
@@ -80,11 +80,12 @@ else
...
 	$(MAKE) db/rollback
-	$(MAKE) db/dump_schema > /tmp/after
-	diff /tmp/{before,after} # No news is good news.
+	$(MAKE) db/dump_schema > /tmp/after_rollback
+	diff /tmp/{before_migration,after_rollback} # No news is good news.
```
Comment (metakuni) [inline backend/Makefile:88]: Renamed the before/after schema dump files.

**Review by atolopko-czi — APPROVED** [review]
LGTM, and thx for the cleanup.  Two ops related questions:
1. Will the migration be run _before_ the code is deployed? Or does do we need to break up the migration and code deploys into separated, ordered PR and deployments?
2. If we revert both the schema and the code, will the code be okay with null values in the restored column?

**Review by Bento007 — APPROVED** [review]

**Review by Bento007 — CHANGES_REQUESTED** [review]
Also update corpora-api.yml

**Comment by metakuni** [comment]
> LGTM, and thx for the cleanup. Two ops related questions:
> 
> 1. Will the migration be run _before_ the code is deployed? Or do we need to break up the migration and code deploys into separated, ordered PR and deployments?

Good question! (to which I did not know the answer 😄 ) It appears that the database migrations run _after_ the `happy` stack is created/updated:
- https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/scripts/happy#L935
- https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/scripts/happy#L987

So, technically, between the time that a stack is created/updated and the migration runs, portal may be _momentarily_ 'broken'. However, this will be the case even if we deploy the migration first in a separate PR followed by deployment of the code changes. The latter will result in a longer potential 'broken' time since more steps are involved.

> 2. If we revert both the schema and the code, will the code be okay with null values in the restored column?

Yes, the dataset artifact's type was never used, so the rollback with nulls should be harmless.

**Review by Bento007 — APPROVED** [review]

**Comment by atolopko-czi** [comment]
> > LGTM, and thx for the cleanup. Two ops related questions:
> > 
> > 1. Will the migration be run _before_ the code is deployed? Or do we need to break up the migration and code deploys into separated, ordered PR and deployments?
>
>  It appears that the database migrations run after the happy stack is created/updated:
>

Good, I think this is the order we want! It would only cause a problem for writes if the column is non-null.  I'm also assuming sqlalchemy is okay with ignoring columns that exist in the db but are not specified in the model. Either way, very short time between code & migration running, so I'm expect we're okay in most any scenario.

> 
> > 2. If we revert both the schema and the code, will the code be okay with null values in the restored column?
> 
> Yes, the dataset artifact's type was never used, so the rollback with nulls should be harmless.

👍

**Pull request merged**